library/std/src/sys/windows/process.rs

```diff
@@ -325,6 +307,84 @@ impl Command {
     }
 }
 
+// Resolve `exe_path` to the executable name.
+//
+// * If the path is simply a file name then use the directories in `PATH` to find the executable.
+// * Otherwise use the `exe_path` as given.
+//
+// This function may also append `.exe` to the name. The rationale for doing so is as follows:
+//
+// It is a very strong convention that Windows executables have the `exe` extension.
+// In Rust, it is common to omit this extension.
+// Therefore this functions first assumes `.exe` was intended.
+// It only falls back to the plain file name if that fails.
+fn resolve_exe<'a>(exe_path: &'a OsStr) -> io::Result<Vec<u16>> {
+    // Early return if there is no filename.
+    if exe_path.is_empty() || path::has_trailing_slash(exe_path) {
+        return Err(io::Error::new_const(
+            io::ErrorKind::InvalidInput,
+            &"program path has no file name",
+        ));
+    }
+    let has_exe_suffix = exe_path.bytes().ends_with(b".exe");
```

Comment:
What about .EXE?

Reply:
I went back and forth on if I should do a case-insensitive match. On the one hand, this is only really a heuristic so it doesn't need to be perfect. And nowadays it seems rare for someone to write the extension in upper case and if they did the only consequence is that `APP.EXE.exe` would be tried before `APP.EXE`. However, now that I think about it again maybe I should cover both cases because there's not really a reason not to.